frontend/web/components/base/SearchableDropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import Icon from 'components/Icon'
```

Comment:
It feels like we need a way for this and EnvironmentSelect to be the same component, they are too similar ?

Reply:
What I had in mind is that this one is just the UI that works with the options and categories directly.
While we have 2 containers using them `RuleConditionPropertySelect` and `EnvironmentSelectDropdown` just in charge of building the options and dealing with the localState.

But actually the most reusable part here is the `GroupLabel` that I could keep here and then just use the `Select` directly in the end wdyt?

Also I noticed that the screenshots are not up to date as we added a custom selection:

<img width="178" height="202" alt="image" src="https://github.com/user-attachments/assets/521d2f2c-ebc9-4e22-8e30-768301b5a4f6" />